src/Report/XmlReport.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Report;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+final class XmlReport extends AbstractReport
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(array $changed)
```

Comment:
@keradus why not? It was just a simple refactoring. I extracted parts of `switch` into separate classes. At other time we can refactor this method. But I think it's too much for a single pr.